package-toolkit/runtime/src/main/kotlin/com/atlan/pkg/cache/DataDomainCache.kt

```diff
@@ -19,6 +19,8 @@ class DataDomainCache(val ctx: PackageContext<*>) : AssetCache<DataDomain>(ctx,
     /** {@inheritDoc} */
     override fun lookupByName(name: String?) {
         // Do nothing: domain cache can only be preloaded en-masse, not retrieved domain-by-domain.
+        val result = lookupByName(name, 0, ctx.client.maxNetworkRetries)
+        if (result != null) cache(result.guid, getIdentityForAsset(result), result)
```

Comment:
As discussed, I would leave this unimplemented and instead do an upfront cache preload if any domain relationships are found in the input file.